x-pack/platform/plugins/shared/ai_infra/product_doc_base/server/services/package_installer/utils/ensure_default_elser_deployed.ts

```diff
@@ -9,7 +9,7 @@ import type { ElasticsearchClient } from '@kbn/core/server';
 import { defaultInferenceEndpoints } from '@kbn/inference-common';
 
 export const ensureDefaultElserDeployed = async ({ client }: { client: ElasticsearchClient }) => {
-  await client.inference.inference(
+  await client.inference.sparseEmbedding(
```

Comment:
[Tests](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/290771/jobs/01961083-273a-4b48-accd-415587aa55d5/artifacts/019610ab-0ac8-4ec0-926b-36558f11adfe) showed that the task type is `sparse_embedding` for this inference_id.

Is this always true, or should we account for other potential `task_types` associated with this `inference_id`?

Reply:
I believe this is fine for now as it is hardcoded to always use ELSER which is a `sparse_embedding` type.
In the future, if this function is expanded to deployed other models this may have to change.